x-pack/plugins/security_solution/common/detection_engine/build_exceptions_query.ts

```diff
@@ -194,15 +210,22 @@ export const buildQueryExceptions = ({
   query,
   language,
   lists,
+  exclude = true,
 }: {
   query: Query;
   language: Language;
-  lists: ExceptionListItemSchema[] | undefined;
+  lists: Array<ExceptionListItemSchema | CreateExceptionListItemSchema> | undefined;
+  exclude?: boolean;
 }): DataQuery[] => {
   if (lists != null) {
-    const exceptions = lists.map((exceptionItem) =>
-      buildExceptionItemEntries({ lists: exceptionItem.entries, language })
-    );
+    const exceptions = lists.reduce((acc, exceptionItem) => {
+      return [
+        ...acc,
+        ...(exceptionItem.entries !== undefined
+          ? [buildExceptionItemEntries({ lists: exceptionItem.entries, language, exclude })]
+          : []),
+      ];
+    }, [] as string[]);
```

Comment:
You should be able to do, `lists.reduce<string>(` and avoid the `as string` cast fwiw.